### PR TITLE
Listen for both close or finish

### PR DESCRIPTION
**Pull request by m64253**
See this: https://github.com/nodejs/node/issues/6929

### CODE REVIEW

File: packages/zipkin-instrumentation-restify/src/restifyMiddleware.js [hunk 1]
```diff
@@ -82,13 +82,19 @@ module.exports = function restifyMiddleware({tracer, serviceName = 'unknown', po
         tracer.recordBinary(Header.Flags, id.flags.toString());
       }
 
-      res.on('finish', () => {
+      const onCloseOrFinish = () => {
+        res.removeListener('close', onCloseOrFinish);
```
Comment (eirslett) [inline packages/zipkin-instrumentation-restify/src/restifyMiddleware.js]: Do we need to remove the listeners? My understanding is that `res.once(...)` will remove them for you, after invoking them?

Comment (m64253) [inline packages/zipkin-instrumentation-restify/src/restifyMiddleware.js]: Yes but the issue is that either `close` or `finish` get triggered so I opted remove both just to be safe.

**Comment by m64253** [comment]
Need help resolving the appvayor erros 😞

**Comment by eirslett** [comment]
Don't worry about the AppVeyor build, it has been broken for some time now. It's due to memcached not being installed on the CI server, which is unrelated to this PR.

**Comment by adriancole** [comment]
thanks for the help!

**Pull request merged**